code/modules/reagents/chemistry/machinery/pandemic.dm

```diff
@@ -158,7 +154,117 @@
 				AD.Refresh()
 		src.updateUsrDialog()
 
+	else if (href_list["eject"])
+		if(beaker)
+			var/obj/item/weapon/reagent_containers/glass/B = beaker
+			B.loc = loc
+			beaker = null
+			icon_state = "mixer0"
+			src.updateUsrDialog()
+			return
+
+	else if (href_list["tab_open"])
+		tab_open = text2num(href_list["tab_open"]) //fucking text
+		src.updateUsrDialog()
+		return
+
+	else if(href_list["chem_choice"])
+		switch(href_list["chem_choice"])
+			if("virusfood")
+				if(virusfood_ammount>0)
+					beaker.reagents.add_reagent("virusfood",max(beaker.reagents.maximum_volume-beaker.reagents.total_volume,1))
+					virusfood_ammount -= 1
+					usr << "Virus Food administered."
+				else
+					usr << "Not enough Virus Food stored!"
+			if("mutagen")
+				if(mutagen_ammount>0)
+					beaker.reagents.add_reagent("mutagen",max(beaker.reagents.maximum_volume-beaker.reagents.total_volume,1))
+					mutagen_ammount -= 1
+					usr << "Unstable Mutagen administered."
+				else
+					usr << "Not enough Unstable Mutagen stored!"
+			if("plasma")
+				if(plasma_ammount>0)
+					beaker.reagents.add_reagent("plasma",max(beaker.reagents.maximum_volume-beaker.reagents.total_volume,1))
+					plasma_ammount -= 1 //no idea why plasma_ammount-- doesn't work here.
+					usr << "Plasma administered."
+				else
+					usr << "Not enough Plasma stored!"
+			if("synaptizine")
+				if(synaptizine_ammount>0)
+					beaker.reagents.add_reagent("synaptizine",max(beaker.reagents.maximum_volume-beaker.reagents.total_volume,1))
+					synaptizine_ammount -= 1
+					usr << "Synaptizine administered."
+				else
+					usr << "Not enough Synaptizine!"
+			if("reset")
+				beaker.reagents.clear_reagents()
+				var/datum/disease/advance/AD = new /datum/disease/advance/inert
+				var/list/data = list("viruses"=list(AD))
+				beaker.reagents.add_reagent("blood",20,data)
+				usr << "Viral strain reset!."
+		src.updateUsrDialog()
+		return
 
+	else if(href_list["update_virus"])
+		if(beaker && beaker.reagents)
+			if(beaker.reagents.reagent_list.len)
+				var/datum/reagent/blood/BL = locate() in beaker.reagents.reagent_list
+				if(BL)
+					if(BL.data && BL.data["viruses"])
+						var/list/viruses = BL.data["viruses"]
+						for(var/datum/disease/D in viruses)
+							var/d_test = 1
+							for(var/datum/disease/DT in new_diseases) //we scan for the desease itself to add to the list
+								if(D.IsSame(DT))
+									d_test = 0
+							if(d_test)
+								new_diseases += D
+								usr << "New disease added to the database!"
+	else if(href_list["update_symptom"])
+		if(beaker && beaker.reagents)
+			if(beaker.reagents.reagent_list.len)
+				var/datum/reagent/blood/BL = locate() in beaker.reagents.reagent_list
+				if(BL)
```

Comment:
this block of four checks is repeated multiple times in this topic call

split it out into it's own proc that returns true if all four checks pass or false if they don’t


    is_valid_beaker  

and call it instead of retyping these four checks over and over

    else if(href_list["update_symptom"])
        if(is_valid_beaker(beaker))
            do thing